### PR TITLE
Disabled dates which exceed 21 days in the past (EXPOSUREAPP-3191)

### PR DESCRIPTION
**Pull request by SamuraiKek**
## Description

This PR fixes a bug where you could select days in the symptom submission calendar which were beyond 21 days in the past.
## How to test
- Get a positive result by scanning a QR code.
- Start the submission of keys process and on the Symptoms screen, select "Yes" and click next
- On the calendar view, the days that are beyond 21 days in the past are greyed out and can't be selected.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/calendar/CalendarDayViewHolder.kt
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/calendar/CalendarDayViewHolder.kt:41]: Maybe also adjust the comment to reflect the new logic.

**Review by harambasicluka — DISMISSED** [review]
lgtm, it just would be nice if you could adjust the comment.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/calendar/CalendarDayViewHolder.kt
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/calendar/CalendarDayViewHolder.kt]: I just saw your comments, but I have adjusted the comments to reflect the new logic in the latest commit.

**Review by harambasicluka — DISMISSED** [review]

**Review by ralfgehrer — CHANGES_REQUESTED** [review]
Screenshot_20201013-142201


Although September 21st cannot be selected, I suppose it should be greyed out as well !?

**Comment by d4rken** [comment]
There is one codesmell according to Sonar, can this be fixed before the merge?

**Comment by SamuraiKek** [comment]
> There is one codesmell according to Sonar, can this be fixed before the merge?
Yep, smelling fresh now.

**Comment by SamuraiKek** [comment]
> Although September 21st cannot be selected, I suppose it should be greyed out as well !?
Past dates are greyed out now. It used to work yesterday but a commit I did today somehow broke it.

**Review by kolyaopahle — APPROVED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/calendar/CalendarDayViewHolder.kt
Comment (kolyaopahle) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/calendar/CalendarDayViewHolder.kt:59]: as a notice for further refactoring, since this is used in multiple places it might make sense to move these checks into a function. Especially if more conditions are introduced in the future.

**Review by kolyaopahle — APPROVED** [review]

**Review by ralfgehrer — APPROVED** [review]
works as expected

**Pull request merged**
